VoodooI2CHID/VoodooI2CMultitouchHIDEventDriver.cpp

```diff
@@ -615,16 +615,24 @@ IOReturn VoodooI2CMultitouchHIDEventDriver::parseElements() {
         if (element->getUsage() == 0)
             continue;
 
-        if (element->conformsTo(kHIDPage_Digitizer, kHIDUsage_Dig_Pen)
-            || element->conformsTo(kHIDPage_Digitizer, kHIDUsage_Dig_TouchScreen)
-            || element->conformsTo(kHIDPage_Digitizer, kHIDUsage_Dig_TouchPad)
-            || element->conformsTo(kHIDPage_Digitizer, kHIDUsage_Dig_DeviceConfiguration)
-            )
-            parseDigitizerElement(element);
-        
-        if (multitouch_interface && element->conformsTo(kHIDPage_Digitizer, kHIDUsage_Dig_TouchScreen))
-            multitouch_interface->setProperty(kIOHIDDisplayIntegratedKey, kOSBooleanTrue);
+        if (forUsage) {
+            if (element->conformsTo(kHIDPage_Digitizer, forUsage)
+                || element->conformsTo(kHIDPage_Digitizer, kHIDUsage_Dig_DeviceConfiguration)
+                )
+                parseDigitizerElement(element);
+        } else {
+            if (element->conformsTo(kHIDPage_Digitizer, kHIDUsage_Dig_Pen)
+                || element->conformsTo(kHIDPage_Digitizer, kHIDUsage_Dig_TouchScreen)
+                || element->conformsTo(kHIDPage_Digitizer, kHIDUsage_Dig_TouchPad)
+                || element->conformsTo(kHIDPage_Digitizer, kHIDUsage_Dig_DeviceConfiguration)
+                )
+                parseDigitizerElement(element);
+            }
+        if (!forUsage || (forUsage == kHIDUsage_Dig_TouchScreen)) {
+            if (multitouch_interface && element->conformsTo(kHIDPage_Digitizer, kHIDUsage_Dig_TouchScreen))
```

Comment:
Why not roll the if statement into one as:
`( *!forUsage || (forUsage == kHIDUsage_Dig_TouchScreen)) && multitouch_interface && element->conformsTo(kHIDPage_Digitizer, kHIDUsage_Dig_TouchScreen))`

Perhaps adding a multi-line condition would be more readable but I am not a fan of nested if statements.